### ISSUE TITLE
NRE in HullIntegrity patches

### ISSUE THREAD

**Issue by IceRaptor** [BattletechModders/CBTBehaviorsEnhanced#33]
Reported in ticket-3435 on RT discord:

```
2020-04-08T16:46:26 FYLS [ERROR] CRITICAL ERROR, PLEASE REPORT:
Delegate OnAttackSequenceEnd - Standard for message type OnAttackSequenceEnd failed with exception 
Object reference not set to an instance of an object
  at CBTBehaviorsEnhanced.HullIntegrity.AttackDirector_OnAttackSequenceEnd+<>c.<ResolveTurretHullBreaches>b__3_0 (BattleTech.MechComponent mc) [0x00006] in <d5abd8016f094a7ab65c144b8f36957c>:0 
  at System.Linq.Enumerable+WhereListIterator`1[TSource].ToList () [0x00017] in <1b13ba6391c74847bbc3eddc86df7eee>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <1b13ba6391c74847bbc3eddc86df7eee>:0 
  at CBTBehaviorsEnhanced.HullIntegrity.AttackDirector_OnAttackSequenceEnd.ResolveTurretHullBreaches (BattleTech.Turret targetTurret) [0x0002d] in <d5abd8016f094a7ab65c144b8f36957c>:0 
```

**Comment by IceRaptor** [comment]
Also

```
2020-04-24T01:38:01 FYLS [ERROR] CRITICAL ERROR, PLEASE REPORT:
Delegate OnAttackSequenceEnd - Standard for message type OnAttackSequenceEnd failed with exception 
Object reference not set to an instance of an object
  at CBTBehaviorsEnhanced.HullIntegrity.AttackDirector_OnAttackSequenceEnd+<>c__DisplayClass4_0.<ResolveVehicleHullBreaches>b__0 (BattleTech.MechComponent mc) [0x00006] in <d5abd8016f094a7ab65c144b8f36957c>:0 
  at System.Linq.Enumerable+WhereListIterator`1[TSource].ToList () [0x00017] in <1b13ba6391c74847bbc3eddc86df7eee>:0 
  at System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x0001f] in <1b13ba6391c74847bbc3eddc86df7eee>:0 
  at CBTBehaviorsEnhanced.HullIntegrity.AttackDirector_OnAttackSequenceEnd.ResolveVehicleHullBreaches (BattleTech.Vehicle targetVehicle) [0x00044] in 
```

**Comment by IceRaptor** [comment]
Implemented various nullity guards and pre-processing to try to avoid these errors. Likely fixed in 0.5.0
